wit/request-reply.wit

```diff
@@ -0,0 +1,27 @@
+/// The request-reply interface allows a guest to send a message and await a response. This
+/// interface is considered optional as not all message services support the concept of
+/// request/reply. However, request/reply is a very common pattern in messaging and as such, we have
+/// included it as a core interface.
+interface request-reply {
+    use types.{client, message, error};
+
+    /// Options for a request/reply operation
+    record request-options {
+        /// The maximum amount of time to wait for a response. If the timeout value is not set, then
+        /// the request/reply operation will block until a message is received in response.
+        timeout-ms: option<u32>,
+        /// The number of expected replies. If the number of replies is not set, then the
+        /// request/reply operation will block until a single message is received in response (or a
+        /// timeout value is reached). This can be used in conjunction for "scatter/gather" type
+        /// operations.
+        expected-replies: option<u32>,
```

Comment:
That feels very specific to a given implementation.

Could the implementation of scatter/gather be possible without it? From the spec perspective, it should be enough as long as I can get multiple replies to a given message.

Reply:
Like I mentioned [here](https://github.com/WebAssembly/wasi-messaging/pull/23#discussion_r1639029618), I'm not totally sold on it, but pushed it up so it would be easier for discussion. I'll resolve this thread here and then we can continue in the other discussion